src/libs/Navigation/CustomActions.js

```diff
@@ -1,6 +1,42 @@
 import {CommonActions, StackActions, DrawerActions} from '@react-navigation/native';
 import lodashGet from 'lodash/get';
 
+/**
+ * Go back to the Main Drawer
+ * @param {Object} navigationRef
+ */
+function navigateBackToDrawer(navigationRef) {
+    let isLeavingDrawerNavigator = false;
+
+    // This should take us to the first view of the modal's stack navigator
+    navigationRef.current.dispatch((state) => {
+        // If this is a nested drawer navigator then we pop the screen and
+        // prevent calling goBack() as it's default behavior is to toggle open the active drawer
+        if (state.type === 'drawer') {
+            isLeavingDrawerNavigator = true;
+            return StackActions.pop();
+        }
```

Comment:
Do we still have nested drawer navigators in use? I think we got rid of the one case. In any case, I think if we want to keep this logic we should make a distinction between a "nested drawer navigator" and a "main" or "root" drawer navigator. 

Otherwise, these variable names are confusing. I understand this code but someone might have a thought like, "Why do we have `isLeavingDrawerNavigator` in a method called `navigateBackToDrawer()`? Shouldn't `isLeavingDrawerNavigator` always be `false`?" Maybe it should be `isLeavingNestedDrawerNavigator` instead.